src/prefect/server/schemas/core.py

```diff
@@ -339,6 +339,13 @@ class TaskRunPolicy(PrefectBaseModel):
     retry_jitter_factor: Optional[float] = Field(
         default=None, description="Determines the amount a retry should jitter"
     )
+    allow_restarts: bool = Field(
+        default=False,
+        description=(
+            "Determines whether retries are allowed after the task has"
+            "reached the RUNNING state."
```

Comment:
This doesn't seem quite accurate. Retries are controlled by the retries option. This allows "backwards" or "same-to-same" transitions?

Reply:
I actually think this should default to `True`

Reply:
We had a lot of problems with this being the default behavior in V1 and the intent of this option was to have people _opt in to possible duplicate runs_ which means they then need to ensure their code is safe for re-execution. I feel like not allowing duplicate runs by default makes more sense since it's clearer orchestrator behavior when you don't have an unstable execution environment.

Reply:
Note this option doesn't affect like... restarting the flow run manually or retries. It's just allowing a run to be orchestrated again when we did not think it should be.

Reply:
I don't think I'm remembering what the v1 problems with this were - what were they?

Reply:
I think primarily user complaints about us not preventing duplicate execution of runs but honestly my memory is vague and I don't have any concrete examples to point to right now 🤷‍♀️